### ISSUE TITLE
Function submit() conflicts with a element with name="submit" 

### ISSUE THREAD

**Issue by Saulis** [PolymerElements/iron-form#122]
Given a simple form:

```html
 <form is="iron-form" method="post">
    <paper-input required></paper-input>
    <input name="submit" type="submit" />
  </form>
```

On function `_onSubmit(e)` there is a call to `this.submit()` and in this case `this.submit` will return the input element. This in turn throws an error, and for example `required` attribute validation does not work on at least Safari.

Since "submit" isn't that uncommon name for the submit button, I'd say this conflict could be beneficial to at least document somewhere.

JSBin: http://jsbin.com/rerazi/edit?html,output (try it on Safari)

**Comment by notwaldorf** [comment]
Yeah, fixing this is hard -- the native `form` has a magical feature where you can "index" elements by their name, so making `this.submit` not refer to the `submit`-named input is impossible.

Would you be interested in sending a PR adding a line to the docs?

**Comment by Saulis** [comment]
Sure, I'll cook up something

**Comment by scmurdock** [comment]
Hey guys. So I have tried really hard to get this to work in Safari. I really love the required feature in iron forms, and could really us it in Safari. 

Here is my original repro for this problem: http://codepen.io/anon/pen/zqaNKd

I changed the name for the submit element: http://codepen.io/anon/pen/PNBENq

Still no dice?
